scipy/stats/_multivariate.py

```diff
@@ -180,6 +225,50 @@ def pinv(self):
         return self._pinv
 
 
+class _CovInfo:
+    """ Class which stores information for a covariance matrix. """
+    def __init__(self, sqrt_cov, sqrt_inv_cov, log_det_cov, rank,
+                 cov=None, inverse_cov=None):
+        """
+        Parameters
+        ----------
+        sqrt_cov: numpy.ndarray
+            An array A such that A @ A.T equals the covariance matrix.
+        sqrt_inv_cov: numpy.ndarray
+            An array A such that A @ A.T equals the inverse of the covariance
+            (or the precision) matrix.
+        log_det_cov: float
+            Log of the determinant of the covariance.
+        cov: numpy.ndarray, optional
+            The actual covariance matrix, if known.
+        inverse_cov: numpy.ndarray, optional
+            The inverse of the covariance matrix, if known.
+
+        """
+        self.sqrt_cov = sqrt_cov
+        self.sqrt_inv_cov = sqrt_inv_cov
+        self.log_det_cov = log_det_cov
+        self.rank = rank
+        self._cov = cov
+        self._inverse_cov = inverse_cov
+
+    @cached_property
```

Comment:
I like the idea of this class. I wonder why it doesn't go a little further. Why not accept all of the information the user can provide in the initializer and store _all of it_ as private attributes in the initializer (e.g. `self._sqrt_cov = _sqrt_cov`), then have public `cached_property`s for all of these attributes that computes them (lazily, in effect) in the most efficient way, given the available information?